### ISSUE TITLE
When I was training, I set imgz=512, but when the model ran one round for automatic validation, the image size was no longer 512, it was a random size. May I ask where I need to modify it so that the training and automatic validation sizes after one round are all fixed at 512

### ISSUE THREAD

**Issue by XiaoBzi** [ultralytics/ultralytics#17874]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train, Val

### Bug

  Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
      1/550       9.6G      3.492      5.365      4.202         63        512:  99%|█████████▉| 92/93 [01:13<00:00,  1.36it/s]cvcvcvcv torch.Size([19, 32, 128, 128])
cvcvcvcv torch.Size([19, 64, 64, 64])
cvcvcvcv torch.Size([19, 128, 32, 32])
cvcvcvcv torch.Size([19, 256, 16, 16])
x_l torch.Size([19, 128, 64, 64])
x_g torch.Size([19, 128, 64, 64])
x_l torch.Size([19, 96, 32, 32])
x_g torch.Size([19, 96, 32, 32])
x_l torch.Size([19, 192, 16, 16])
x_g torch.Size([19, 192, 16, 16])
/root/anaconda3/envs/yolov5/lib/python3.9/site-packages/torch/autograd/__init__.py:200: UserWarning: upsample_bilinear2d_backward_out_cuda does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at ../aten/src/ATen/Context.cpp:71.)
  Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
/root/anaconda3/envs/yolov5/lib/python3.9/site-packages/torch/autograd/__init__.py:200: UserWarning: adaptive_avg_pool2d_backward_cuda does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at ../aten/src/ATen/Context.cpp:71.)
  Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
/root/anaconda3/envs/yolov5/lib/python3.9/site-packages/torch/autograd/__init__.py:200: UserWarning: reflection_pad2d_backward_cuda does not have a deterministic implementation, but you set 'torch.use_deterministic_algorithms(True, warn_only=True)'. You can file an issue at https://github.com/pytorch/pytorch/issues to help us prioritize adding deterministic support for this operation. (Triggered internally at ../aten/src/ATen/Context.cpp:71.)
  Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
      1/550       9.6G      3.491      5.365      4.202         30        512: 100%|██████████| 93/93 [01:14<00:00,  1.24it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95):   0%|          | 0/14 [00:00<?, ?it/s]cvcvcvcv **torch.Size([64, 32, 136, 136])
cvcvcvcv torch.Size([64, 64, 68, 68])
cvcvcvcv torch.Size([64, 128, 34, 34])
cvcvcvcv torch.Size([64, 256, 17, 17])**
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95):   0%|          | 0/14 [00:00<?, ?it/s]

### Environment

I can train the medel,there is no problem about the environment.

### Minimal Reproducible Example

 this is train process
 Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
      1/550       9.6G      3.492      5.365      4.202         63        512:  99%|█████████▉| 92/93 [01:13<00:00,  1.36it/s]cvcvcvcv torch.Size([19, 32, 128, 128])
cvcvcvcv torch.Size([19, 64, 64, 64])
cvcvcvcv torch.Size([19, 128, 32, 32])
cvcvcvcv torch.Size([19, 256, 16, 16])
x_l torch.Size([19, 128, 64, 64])
x_g torch.Size([19, 128, 64, 64])
x_l torch.Size([19, 96, 32, 32])
x_g torch.Size([19, 96, 32, 32])
x_l torch.Size([19, 192, 16, 16])
x_g torch.Size([19, 192, 16, 16])
this is auto val process after train
      1/550       9.6G      3.491      5.365      4.202         30        512: 100%|██████████| 93/93 [01:14<00:00,  1.24it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95):   0%|          | 0/14 [00:00<?, ?it/s]cvcvcvcv torch.Size([64, 32, 136, 136])
cvcvcvcv torch.Size([64, 64, 68, 68])
cvcvcvcv torch.Size([64, 128, 34, 34])
cvcvcvcv torch.Size([64, 256, 17, 17])
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95):   0%|          | 0/14 [00:00<?, ?it/s]

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @XiaoBzi, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com) for guidance, where many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples are available to help troubleshoot or answer common questions.

It looks like you may have encountered a specific issue regarding different image sizes during training and validation. If this is a 🐛 Bug Report, we kindly request a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) outlining the exact steps that lead to the behavior. This would help us better understand and address your problem.

In the meantime, please ensure the following:

## Upgrade

Confirm you're using the latest version of the `ultralytics` package, including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), within a [**Python>=3.8**](https://www.python.org/) environment and with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) installed. Upgrade with:

```bash
pip install -U ultralytics
```

## Environments

You can test your training setup within various tested environments, such as:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

If your issue persists after these steps, please provide more details including any relevant portions of your code, configuration, and logs to help us reproduce the behavior. Additionally, verify that you have reviewed your training configuration and ensured correct parameter usage, including `imgsz`, via our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

## Join the Community

For real-time support, you can join the [Ultralytics Discord](https://discord.com/invite/ultralytics) 🎧. If you prefer detailed discussions, explore our [Discourse Forum](https://community.ultralytics.com). The [Ultralytics Subreddit](https://reddit.com/r/Ultralytics) is also a great place to engage in community conversations and share insights.

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, it means all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests verify the correct operation of YOLO's [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu environments.

P.S. This is an automated response to guide you in addressing your issue efficiently. An Ultralytics engineer will review and follow up with further assistance soon 🙂

**Comment by Y-T-G** [comment]
You can pass `deterministic=False` to `model.train()`

**Comment by Y-T-G** [comment]
What model is this? RTDETR?
